core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/CompactEntityComponent.scala

```diff
@@ -526,6 +526,94 @@ class CompactEntityQuery(driverComponent: DriverComponent)
   ): SqlStreamingAction[Seq[Entity], Entity, Read] =
     queryEntitiesWithFilter(workspaceId, entityType, entityQuery, sql"")
 
+  /**
+   * Renames an entity in the ENTITY table.
+   *
+   * Returns the number of entities that were renamed.
+   *
+   * `execution plan: index range scan on idx_entity_type_name`
+   */
+  def renameEntity(workspaceId: UUID, entityType: String, oldName: String, newName: String): ReadWriteAction[Int] = {
+    // Update the entity name in the ENTITY table
+    // explain plan: index range scan on idx_entity_type_name
+    val updateEntityNameSql = sql"""update ENTITY set name = $newName, record_version = record_version + 1
+          where workspace_id = $workspaceId
+          and entity_type = $entityType
+          and name = $oldName
+          and deleted = 0"""
+
+    // Update the entity from_name in the ENTITY_REFS table
+    // explain plan: index range scan on unq_from_to
+    val updateFromNameSql =
+      sql"""update ENTITY_REFS
+            set from_name = $newName
+            where workspace_id = $workspaceId
+            and from_entity_type = $entityType
+            and from_name = $oldName"""
+
+    // Update the entity to_name in the ENTITY_REFS table
+    // explain plan: index range scan on unq_from_to
+    val updateToNameSql =
+      sql"""update ENTITY_REFS
+            set to_name = $newName
+            where workspace_id = $workspaceId
+            and to_entity_type = $entityType
+            and to_name = $oldName"""
+
+    // Get all paths in attributes that reference the old name
+    // explain plan: non-unique index scan on idx_entity_type_name and idx_to
+    val attrRefRegex = """'\\$\\.attrs\\.[^.]+\\.entityName'"""
```

Comment:
nit: use `CompactEntitySerialization.ATTRS_KEY` instead of hardcoding "attrs" and `AttributeFormat.ENTITY_NAME_KEY` instead of hardcoding "entityName"